packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-auth-userpool-fields.ts

```diff
@@ -0,0 +1,1925 @@
+      const modelName = 'TodoCustomGroupContentVarious';
+      const modelOperationHelpersAdmin = createModelOperationHelpers(appSyncClients[userPoolProvider][userName1], schema);
+      const modelOperationHelpersNonAdmin = createModelOperationHelpers(appSyncClients[userPoolProvider][userName2], schema);
+      const todoHelperAdmin = modelOperationHelpersAdmin[modelName];
+      const todoHelperNonAdmin = modelOperationHelpersNonAdmin[modelName];
+
+      const todoWithAllowedFields = {
+        privateContent: 'PrivateContent',
+        customGroup: adminGroupName,
+      };
+      const todo = {
+        ...todoWithAllowedFields,
+        publicContent: 'PublicContent',
+      };
+      const completeResultSet = `
+        id
+        customGroup
+        privateContent
+        publicContent
+      `;
+      const createResultSetName = `create${modelName}`;
+
+      await expect(async () => await todoHelperAdmin.create(createResultSetName, todo)).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"GraphQL error: Not Authorized to access create${modelName} on type Mutation"`,
+      );
+
+      // Create a record with allowed fields only, so we can test the update and delete operations.
+      const allowedResultSet = `
+        id
+        customGroup
+        privateContent
+      `;
+      const createResult = await todoHelperAdmin.create(createResultSetName, todoWithAllowedFields, allowedResultSet);
+      expect(createResult.data[createResultSetName].id).toBeDefined();
+      todo['id'] = createResult.data[createResultSetName].id;
+      expect(createResult.data[createResultSetName].privateContent).toBeNull();
+      expect(createResult.data[createResultSetName].customGroup).toBeNull();
+
+      const todoUpdated = {
+        ...todo,
+        privateContent: 'Private Content updated',
+        publicContent: 'Public Content updated',
+      };
+
+      await expect(async () => await todoHelperAdmin.update(`update${modelName}`, todoUpdated)).rejects.toThrowErrorMatchingInlineSnapshot(
+        `"GraphQL error: Not Authorized to access update${modelName} on type Mutation"`,
+      );
+
+      const expectedFieldError = `"GraphQL error: Not Authorized to access publicContent on type ${modelName}"`;
+      const getResult = await todoHelperAdmin.get({ id: todo['id'] }, undefined, true, 'all');
+      checkOperationResult(getResult, { ...todo, publicContent: null }, `get${modelName}`, false, [expectedFieldError]);
+
+      const listTodosResult = await todoHelperAdmin.list({}, completeResultSet, `list${modelName}`, false, 'all');
+      checkListItemExistence(listTodosResult, `list${modelName}`, todo['id'], true);
+      checkListResponseErrors(listTodosResult, [`Not Authorized to access publicContent on type ${modelName}`]);
+    });
+
+    test.skip('user not part of group in allowed custom groups list of a record can perform CRUD and subscription operations with only allowed fields in selection set', async () => {
```

Comment:
As above, add an easily searchable todo, or link to this code in our backlog ticket